cpp/src/arrow/csv/reader.cc

```diff
@@ -339,12 +343,9 @@ class SerialTableReader : public BaseTableReader {
   using BaseTableReader::BaseTableReader;
 
   Status Init() override {
-    RETURN_NOT_OK(
-        io::MakeInputStreamIterator(input_, read_options_.block_size, &block_iterator_));
     // Since we're converting serially, no need to readahead more than one block
-    int block_queue_size = 1;
-    return MakeReadaheadIterator(std::move(block_iterator_), block_queue_size,
-                                 &block_iterator_);
+    return io::MakeInputStreamIterator(input_, read_options_.block_size)
```

Comment:
Not sure why you removed the readahead iterator here?

Reply:
when reading serially, is it useful to have a single element block queue?

Reply:
It means you're reading the next block ahead while parsing the current block. One day we'll need to run some tests on remote filesystems :-)

Reply:
I'll revert this, then